BUG: Make Series flex methods raise NotImplementedError for bool dtypes

majiayu000 opened the pull request:
## Summary

This PR fixes the inconsistency between Series flex methods (`.truediv()`, `.floordiv()`, etc.) and their dunder counterparts (`__truediv__`, `__floordiv__`, etc.) when operating on boolean dtypes.

Previously:
- `pd.Series([True]) / [True]` raised `NotImplementedError` ✓
- `pd.Series([True]).truediv([True])` returned a float result ✗

After this fix, both methods consistently raise `NotImplementedError: operator 'truediv' not implemented for bool dtypes`.

## Changes

Modified `Series._binop()` to route through `ops.arithmetic_op()` instead of directly calling the operator function. This ensures the `_bool_arith_check` is applied consistently for all arithmetic operations.

The fix applies to all disallowed operations for bool dtypes:
- `truediv` / `rtruediv`
- `floordiv` / `rfloordiv`  
- `pow` / `rpow`
- `divmod` / `rdivmod`

## Test Plan

- [x] Verified `pd.Series([True]).truediv([True])` now raises `NotImplementedError`
- [x] Verified `pd.Series([True]).rtruediv([True])` now raises `NotImplementedError`
- [x] Verified `pd.Series([True]).floordiv([True])` now raises `NotImplementedError`
- [x] Verified `pd.Series([True]).pow([True])` now raises `NotImplementedError`
- [x] Verified normal float operations still work: `pd.Series([1.0]).truediv([2.0])` returns `[0.5]`

Closes #63250

jbrockmendel commented on pandas/core/series.py:
```diff
@@ -7060,7 +7060,26 @@ def _binop(self, other: Series, func, level=None, fill_value=None) -> Series:
+                    f"operator '{op_name}' not implemented for bool dtypes, "
+                    f"it will raise in a future version. "
+                    f"Use .astype(int) before calling .{func.__name__}().",
+                    FutureWarning,
```
Pandas4Warning

jbrockmendel commented on pandas/core/series.py:
```diff
@@ -7060,7 +7060,26 @@ def _binop(self, other: Series, func, level=None, fill_value=None) -> Series:
+            # in a future version.
+            if (
+                func in ops.array_ops._BOOL_OP_NOT_ALLOWED
+                and hasattr(this_vals, "dtype")
```
doesnt this_vals aways have dtype?

jbrockmendel replied:
i think we also want to check other_vals?

majiayu000 committed:
BUG: Deprecate bool dtype Series flex ops (#63250)

Emit Pandas4Warning when flex arithmetic methods (truediv, floordiv,
pow, divmod and their reverse variants) are called on bool dtypes.
The dunder counterparts already raise NotImplementedError; flex methods
will do the same in a future version.

Signed-off-by: majiayu000 <1835304752@qq.com>

jbrockmendel commented:
Closing as stale. can reopen when you're ready to address comments and continue.
